lib/template/eval.c

```diff
@@ -117,7 +120,14 @@ log_template_append_elem_func(LogTemplate *self, LogTemplateElem *e, LogTemplate
    */
   if (e->func.ops->eval)
     e->func.ops->eval(e->func.ops, e->func.state, &args);
-  e->func.ops->call(e->func.ops, e->func.state, &args, result, &value_type);
+  e->func.ops->call(e->func.ops, e->func.state, &args, target_buffer, &value_type);
+
+  if (escape)
```

Comment:
This commit `template: perform escaping on the result of functions` is a breaking change. Shouldn't we support the old behavior for configs with `@version 4.4` and below?

Reply:
We decided to not support the old behavior because we see minimal chance that someone used `template-escape(yes)` with multi-level template functions. `template-escape()` was only available for writer based dest drivers, and its original use case was to workaround not having sql destination, which we do have now.

We will mention this in the newsfile though.

Reply:
See #4679.